pkg/agent/waitfor.go

```diff
@@ -0,0 +1,107 @@
+package agent
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// WaitFor wait for the installation complete triggered by the agent installer.
+func WaitFor() error {
+	logrus.Info("WaitFor command")
+
+	// ctx := context.Background()
+	// zeroClient, err := NewNodeZeroClient()
+	// if err != nil {
+	// 	return err
+	// }
+
+	return nil
+}
+
+func WaitForNodeZeroAgentRestAPIInit(ctx context.Context, zeroClient *nodeZeroClient, timeoutMins int) error {
```

Comment:
🤔 To expand this thought I wonder if it is NodeZeroChecker, ClusterZeroChecker or both?

`NodeZeroChecker` would be about nodeZero. The first half of `waitfor bootstrap` would be under this because we are inspecting nodeZero mostly. After nodeZero reboots we do a couple checks against the cluster api to validate bootstrap is complete.
`ClusterZeroChecker` would be about clusterZero which could include the `isKubeAPILive` and `doesKubeConfigExist` methods. This would also clarify that we are talking about the cluster we are installing, clusterZero.

Reply:
or maybe these are on type `clusterZero` and the checkers are another layer of abstraction. 🤔 I will fiddle with it today